web/src/filter.ts

```diff
@@ -1479,6 +1479,11 @@ export class Filter {
                 case "in-all":
                     return $t({defaultMessage: "All messages including muted channels"});
                 case "channels-public":
+                    if (page_params.is_spectator) {
```

Comment:
I think the right check is whether you're a spectator or guest. (I think we treat spectators as guests).